aicsimageio/writers/ome_tiff_writer.py

```diff
@@ -0,0 +1,641 @@
+            )
+            photometric = (
+                TIFF.PHOTOMETRIC.RGB if is_rgb else TIFF.PHOTOMETRIC.MINISBLACK
+            )
+            planarconfig = TIFF.PLANARCONFIG.CONTIG if is_rgb else None
+            tif.write(
+                data[scene_index],
+                description=description,
+                photometric=photometric,
+                metadata=None,
+                planarconfig=planarconfig,
+                compression=TIFF.COMPRESSION.ADOBE_DEFLATE,
+            )
+
+        tif.close()
+
+    @staticmethod
+    def _resolve_OME_dimension_order(
+        shape: Tuple[int, ...], dimension_order: Union[str, None]
+    ) -> Tuple[str, bool]:
+        """
+        Do some dimension validation and return an ome-compatible 5D dimension order
+        and whether the data is rgb multisample
+
+        Parameters
+        ----------
+        shape: Tuple[int, ...]
+            A data array shape
+        dimension_order: Union[str, None]
+            A dimension order string, composed of some subset of TCZYXS
+
+        Returns
+        -------
+        Tuple[str, bool]
+            An OME-compatible 5D dimension_order string and a boolean for whether the
+            data shape had rgb samples
+        """
+        ndims = len(shape)
+
+        if ndims > 5 and (shape[-1] != 3 and shape[-1] != 4):
+            raise ValueError(
+                f"Passed in greater than 5D data but last dimension is not 3 or 4: "
+                f"{shape[-1]}"
+            )
+
+        if dimension_order is not None and len(dimension_order) != ndims:
+            raise InvalidDimensionOrderingError(
+                f"Dimension order string has {len(dimension_order)} dims but data "
+                f"shape has {ndims} dims"
+            )
+
+        # data is rgb if last dimension is S and its size is 3 or 4
+        is_rgb = False
+        if dimension_order is None:
+            # we will only guess rgb here if ndims > 5
+            # I could make a better guess if I look at any ome-xml passed in
+            is_rgb = ndims > 5 and (shape[-1] == 3 or shape[-1] == 4)
+            dimension_order = (
+                DEFAULT_DIMENSION_ORDER_WITH_SAMPLES
+                if is_rgb
+                else DEFAULT_DIMENSION_ORDER
+            )
+        else:
+            is_rgb = dimension_order[-1] == "S" and (shape[-1] == 3 or shape[-1] == 4)
+
+        # select last 5 (or 6 if rgb) dims
+        max_dims = 6 if is_rgb else 5
+        if ndims > max_dims:
+            raise ValueError(
+                f"Data array should be less than 7D: is_rgb = {is_rgb} and {shape}"
+            )
+        #     slc = [0] * (ndims - max_dims)
+        #     slc += [slice(None)] * max_dims
+        #     data = data[slc]
+
+        ndims = len(shape)
+        assert (
+            ndims == 6 or ndims == 5 or ndims == 4 or ndims == 3 or ndims == 2
+        ), "Expected no more than 6 dimensions in data array"
```

Comment:
Didn't you just do this check above?

Reply:
it's not exactly the same check, one is for rgb and the other is for absolute number of dims.  i can make it clearer.